lib/linguist/vendor.yml

```diff
@@ -87,8 +87,8 @@
 - (^|/)select2/.*\.(css|scss|js)$
 
 # Vendored dependencies
-- third[-_]?party/
-- 3rd[-_]?party/
+- third([-_]?p|P)arty/
+- 3rd([-_]?p|P)arty/
```

Comment:
These will match `3rd-party`, but not `3rd-Party`. The expressions need to be:

~~~regexp
- third[-_]?[Pp]arty/
- 3rd[-_]?[Pp]arty/
~~~

Come to think of it, they can both be sensibly collapsed into one expression:

~~~regexp
- (3rd|[Tt]hird)[-_]?[Pp]arty/
~~~

Reply:
> These will match `3rd-party`, but not `3rd-Party`.

Yes, this was intentional as I wanted to do a minimal change. I force-pushed with your suggested combined regular expression which definitely makes sense.